process/block/preprocess/transactions.go

```diff
@@ -545,24 +597,10 @@ func (txs *transactions) CreateAndProcessMiniBlock(
 			continue
 		}
 
-		currTxGasLimit := txs.economicsFee.ComputeGasLimit(orderedTxs[index])
-		if core.IsSmartContractAddress(orderedTxs[index].RcvAddr) {
-			currTxGasLimit = orderedTxs[index].GasLimit
-		}
-
-		isGasLimitReached := addedGasLimitPerCrossShardMiniblock+currTxGasLimit > process.MaxGasLimitPerMiniBlock
-		if isGasLimitReached {
-			log.Trace("max gas limit per mini block is reached",
-				"num added txs", len(miniBlock.TxHashes),
-				"ordered txs", len(orderedTxs),
-			)
-			continue
-		}
-
 		snapshot := txs.accounts.JournalLen()
 
 		// execute transaction to change the trie root hash
-		err := txs.processAndRemoveBadTransaction(
+		err = txs.processAndRemoveBadTransaction(
```

Comment:
you should not start processing a transaction if gasLimitIsReached is close to be reached, and the current tx has a big gaslimit. You calculate before, and take a decision if you may introduce the new transaction, if yes then process. If gasLimit is 10000 and the current added gas is 9900, it does not make sense to try to process a tx with 5000 gas.

Reply:
I did it in this way because it could be an edge case, if let's say this tx is a smart contract transaction with sender and receiver in self shard and let say that it would have 4900 refunded gas. In this case it could be included in this mini block. If I would made this check before I would skipped it.